BlueDBOnDisk/src/main/java/io/bluedb/disk/recovery/RecoveryManager.java

```diff
@@ -71,4 +123,24 @@ public void recover() {
 			}
 		}
 	}
+
+	public static Long extractTimestamp(File file) {
+		try {
+			String fileName = file.getName();
+			String longString = fileName.split("[.]")[0];
+			return Long.valueOf(longString);
+		} catch (Throwable t) {
+			return null;

```

Comment:
returning null here can have some bad consequences. If we use the primitive without checking for `null` in the boxed value, we can get an NPE that wrecks us. 

We are occasionally checking this value with a `<` without checking it. I'd recommend returning some marker value, like -1. Or, failing that, we might go crazy and return an Optional<Long>.

Reply:
so I filtered out the ones with null timestamps.  if you don't like the change, maybe we should discuss